### ISSUE TITLE
Div operator for path concatenation

### ISSUE THREAD

**Issue by theiviaxx** [nemec/pathlib#11]
This is such a cool and useful part of pathlib (python).  I know its probably against any standard out there, but i always make an exception for this feature.  Anyhow, that seems to be the only thing missing from this lib.

Thanks!

**Comment by nemec** [comment]
Thanks for the kind words. I debated whether or not to add that feature while I was building the library, although I decided against it in the end because, as you mention, it is kind of out of character for the .NET world.

That said, it's been a few years and I think I might revisit that decision. It's certainly a handy feature :)

**Comment by ericpohl** [comment]
I would add a +1 for this feature. I think the .NET community has evolved toward "terse and easy" over the years - one example is the [new console app template](https://docs.microsoft.com/en-us/dotnet/core/tutorials/top-level-templates). I'd happily submit a PR for it if this project is still being maintained.

**Comment by nemec** [comment]
Hey there, I don't know if y'all still have any interest in this project, but I just published version 0.9.7.1 with support for this. Only thing to note is that on .NET Framework (legacy) you cannot join this way on one of the interfaces (`IPurePath`/`IPath`) since interfaces didn't support operators back then. If you're on .NET standard 2.1 or newer it should work though.

```csharp
var path = PathLib.Paths.Create("~/prg");
Console.WriteLine(path / "pathlib");
```

```csharp
var path = Paths.Create("~/prg");
var path2 = Paths.Create("pathlib");
Console.WriteLine(path / path2);
```

.NET Framework
```csharp
var path = (WindowsPath)Paths.Create("~/prg");
Console.WriteLine(path / "pathlib");
```

**Comment by theiviaxx** [comment]
awesome, yeah i'll give it a go.  i have a new project that could use this 👍
